src/Compilers/VisualBasic/Portable/Syntax/SyntaxFactory.vb

```diff
@@ -532,5 +532,91 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return SyntaxEquivalence.AreEquivalent(oldList.Node, newList.Node, ignoreChildNode, topLevel:=False)
         End Function
 
+        ''' <summary>
+        ''' Determines if a syntax node is a part of a LINQ query.
+        ''' If so then a query is not terminated the syntax tree ends with an empty line.
+        ''' </summary>
+        ''' <returns>True if the token is part of a LINQ query and syntax tree ends with an empty line.</returns>
+        Private Shared Function IsPartOfUnterminatedLinqQuery(token As SyntaxToken, statementNode As SyntaxNode, endOfFileToken As SyntaxToken) As Boolean
+            Dim node = token.Parent
+            Do
+                If node.IsKind(SyntaxKind.QueryExpression) Then
+                    Return Not endOfFileToken.LeadingTrivia.Contains(Function(trivia) trivia.IsKind(SyntaxKind.EndOfLineTrivia))
```

Comment:
Would it make sense to check this condition on the ```endOfFileToken``` before traversing the syntax?